upconsulting/IsisCB
# Python3 (#653)

* ran futurize


Former-commit-id: 174be9dc26b722f1f14206697faea7f22a9a4d0f

* removed sql file


Former-commit-id: a572068ed0e56ca6c10ba29f6ea394deb6e707ed

* checked on_delete values for model file

* more on_delete checks

* ported to python, server runs but still need to manually add on_delete to migrations

* run travis with python 3

* updated python version for travis

* change for travis

* removed dependency that couldn't be installed (shouldn't be there, it's an ubuntu/debian bug)

* fixed (some?) test settings

* test_settings and null property on foreinkey model properties

* added on deletes

* removed hotshot, django_pagination, SessionAuth middleware and added dj_pagination

* changed static tag

* fixed some python3 issues

* add try/except to avoid issue with migration on new db

* change test db name

* update python version in dockerfile

* adjustments to production settings

* update postgres verison in yum

* added postgres rpm

* changed rpm link

* removed rpm, added amazon extras

* updated postgresql package

* move postgresql install to script

* changed mod_ssl version

* let's see if we need all these things

* install requirements first

* changed order

* renamed middleware property

* update storage for staticfile

* we might not need postgres?

* Adjust venv path to amazon linux

* remove amazon-linux-extras

* revert mod-ssl version back

* update pip and install requirement in commands

* update s3 configurations

* remove storages from apps

* remove old storages libs?

* hm...

* yes to uninstall

* add storages back into installed apps

* remove storages again

* revert everything back and try to get Amazon Linux 2 to run

* update venv path

* adjust wsgi path

* get gunicorn to work

* fix procfile...

* install requirements in dockerfile

* install requirements before descending in directory

* adjusted dockerfile

* more adjustements

* add expose directive for AWS

* trying to stop nginx from failing

* set no proxy server for docker

* adjust copy directive

* use python 3.6 due to celery

* set up celery results

* fix queue prefix for celery

* revert celery change

* Revert "revert celery change"

This reverts commit f5c953c4cdac4988249050ee7f9247814112e2aa.

* remove proxy server configuration

* trying to get celery to run

* remove queue prefix?

* let's see if the routing key fixes things

* set celery log level to debug

* update travis to deploy python3 to AWS

* trying to fix autodeployment

* move back to parent directory for deploy

* undo changes of procfile

* set celery back to debug

* let's see if its a celery issue or not

* put back date condition for timelines; loglevel back to info

* made login work for local accounts; some bugfixes

* added site_id

* created migrations

* Fixed logout and sign up

* some bugfixes for admin site; started on social login;

* Enabled Twitter login and password change page

* fixed some issues with encoding and bulk change reindex

* fixed citation view

* remove print

* fixed authority searching for subjects/categories; added __str__ function

* [IEXP-193] fixed citation collection list view

* [IEXP-192] fixed user list

* Fixed a javascript issue with quotation marks

* Update users.html

* [IEXP-188, IEXP-194, IEXP-195] fixes to zotero pages, added missing str function

* [IEXP-188, IEXP-190] create default user profile for third party accoutns

* [IEXP-188, IEXP-191] add password set page

* Added facebook integration

* added authority reindexing action; nginx conf for redirect

* move nginx conf file

* move nginx config to config file

* fix indentation

* Fixed authority bulk change bug

* [IEXP-208] misconfiguration in urls broke rest api

* [IEXP-205] added __str__ method for string repr

* print to check what's up with timeline creation

* [IEXP-205] use str instead of string; [IEXP-210] logger output

* [IEXP-211] Changed related institution modal header

* [IEXP-212] log statements

* [IEXP-212] adding logger

* don't change file permissions

* trying to get permissions right and [IEXP-216]

* [IEXP-212] use db cache

* [IEXP-220] catch object not found exception

* [IEXP-215] open files in correct mode

* [IEXP-220] fix record type condition

* [IEXP-218] added filter object back into context

* [IEXP-221, IEXP-227] added classification code; fixed export

* [IEXP-222] added str method for citations collections

* [IEXP-223] added str method to IsiscbRole

* [IEXP-224] adjust allauth registration

* [IEXP-224] fix email integration for registration and password reset

* [IEXP-229] fix writing to csv writer; added from email address in settings

* [IEXP-224, IEXP-226] fixing email settings

* [IEXP-224, IEXP-226] fixed admin widgets to set site name

* [IEXP-213] fix quotation mark issue

* [IEXP-214] submit correct form to make sorting work

* [IEXP-232] updated task to check if field is many to many relation

if so, add new value rather than replace as it was done before; also made upload path configurable for update element and removed old settings files

* [IEXP-233] removed b option from file opening so csv package doesn't get bytes

* [IEXP-222] fixed authority collection issues

* [IEXP-225] Added template for signup

* [IEXP-225] created management command to migrate old social auth accounts to allauth

* [IEXP-229, IEXP-175] added fix from IEXP-175 to fix issue in IEXP-229

* [OAuth] added authentication error template and auto signup

* [IEXP-225] Added note about social login issues

added connections template

* remove backup files; enable test for travis

* remove logging handler

* fix db name in travis

* run migrate before tests

* added elasticsearch to travis

* adjusted travis for new production and staging deploy

Co-authored-by: Taylor Quinn <tquinn2035@gmail.com>

[IEXP-193]: https://upconsulting.atlassian.net/browse/IEXP-193
